fundingmanager.go

```diff
@@ -1212,18 +1221,33 @@ func (f *fundingManager) handleFundingOpen(fmsg *fundingOpenMsg) {
 	)
 	tweaklessCommitment := localTweakless && remoteTweakless
 	chainHash := chainhash.Hash(msg.ChainHash)
+
+	// If the peer sent an upfront shutdown script, but we do not support the
+	// feature, log a warning. Enforcement of this script occurs entirely on
+	// our side, so we can ignore it.
+	localUpfrontShutdown := fmsg.peer.LocalLocalFeatures().HasFeature(
+		lnwire.UpfrontShutdownScript,
+	)
+	if !localUpfrontShutdown && len(msg.UpfrontShutdownScript) != 0 {
+		fndgLog.Warnf("Peer: %x sent upfront shutdown address: %x, we do "+
+			"not support the feature", fmsg.peer.PubKey(),
+			msg.UpfrontShutdownScript)
+	}
+
 	req := &lnwallet.InitFundingReserveMsg{
-		ChainHash:        &chainHash,
-		NodeID:           fmsg.peer.IdentityKey(),
-		NodeAddr:         fmsg.peer.Address(),
-		LocalFundingAmt:  0,
-		RemoteFundingAmt: amt,
-		CommitFeePerKw:   lnwallet.SatPerKWeight(msg.FeePerKiloWeight),
-		FundingFeePerKw:  0,
-		PushMSat:         msg.PushAmount,
-		Flags:            msg.ChannelFlags,
-		MinConfs:         1,
-		Tweakless:        tweaklessCommitment,
+		ChainHash:             &chainHash,
+		NodeID:                fmsg.peer.IdentityKey(),
+		NodeAddr:              fmsg.peer.Address(),
+		LocalFundingAmt:       0,
+		RemoteFundingAmt:      amt,
+		CommitFeePerKw:        lnwallet.SatPerKWeight(msg.FeePerKiloWeight),
+		FundingFeePerKw:       0,
+		PushMSat:              msg.PushAmount,
+		Flags:                 msg.ChannelFlags,
+		MinConfs:              1,
+		Tweakless:             tweaklessCommitment,
+		LocalUpfrontShutdown:  shutdown,
```

Comment:
how do we handle the case where these are nil?

Reply:
A nil `DeliveryAddress` is just an empty byte array, and there are checks for `len!=0` when we do use it.